### ISSUE TITLE
Allow setting `RUST_FLAGS` for "Build: Sync"

### ISSUE THREAD

**Issue by jkosh44** [intellij-rust/intellij-rust#10665]
## Environment

* **IntelliJ Rust plugin version:** 0.4.198.5409-231
* **Rust toolchain version:** rustc 1.68.2 (9eb3afe9e 2023-03-27)
* **IDE name and version:** CLion 22023.1.4
* **Operating system:** Ubuntu 22.04.2

## Problem description

The "Build: Sync" process happens every time CLion is opened, which attempts to compile and check the entire project. There is no way to set the `RUST_FLAGS` environment variable used during this compilation. If a user builds there project with `RUST_FLAGS` set to a different value than "Build: Sync", then "Build: Sync" will recompile from scratch and next time the user compiles their project they will have to recompile from scratch.

## Steps to reproduce

1. Build your project with `RUST_FLAGS` set to something non-default such as `RUSTFLAGS=-C link-arg=-fuse-ld=lld;`.
2. Restart Your IDE. Note that "Build: Sync" is recompiling from scratch.
3. Build your project again with `RUST_FLAGS` set. Note that it is recompiling from scratch.

**Comment by Maksim20023** [comment]
Hi! Thank you for reaching us.
To reproduce the issue more accurately, could you please specify how you set the environment variables?

**Comment by jkosh44** [comment]
Yes, I specify run variables via the Run/Debug configurations. See attached
![Screenshot from 2023-07-14 10-51-47](https://github.com/intellij-rust/intellij-rust/assets/10800303/58921a88-8c93-4d8b-8a37-56a80b9db18b)

**Comment by PatricioNapoli** [comment]
Hello, bumping this issue @Maksim20023 

The cargo project 'Sync' that CLion performs, does not let you define custom env vars or arguments.

For env vars, you have to set them in user's home .profile, .bashrc or .zshrc (I added my env vars to all of them just in case CLion acted funny)

This is different from the run configurations, those "mostly" work, except for running clippy, that one fails due to CLion injecting 'message-format' argument.

**Comment by jkosh44** [comment]
Thanks for the tip @PatricioNapoli, unfortunately that doesn't seem to work for me.

**Comment by Maksim20023** [comment]
Hello @jkosh44 ! Could you please try to move your environment variables from run\debug configuration to [config.toml](https://doc.rust-lang.org/cargo/reference/config.html) file and observe if the issue still persists. If not, could this workaround be convenient for you?

**Comment by jkosh44** [comment]
That doesn't seem to work either.

**Comment by jkosh44** [comment]
Starting CLion from the command line with the environment variable specified seems to work for me.

```
$ RUSTFLAGS="-C link-arg=-fuse-ld=lld" clion
```

**Comment by Maksim20023** [comment]
Hi @jkosh44  
Thank you for sharing your experience. Unfortunately, we currently don't have ETA for the issue. We will add a reply when we have any new information.
